src/v/datalake/protobuf_to_arrow_converter.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+#include "datalake/errors.h"
+#include "datalake/logger.h"
+#include "datalake/proto_to_arrow_interface.h"
+#include "datalake/proto_to_arrow_scalar.h"
+#include "datalake/proto_to_arrow_struct.h"
+
+#include <arrow/api.h>
+#include <arrow/array/builder_base.h>
+#include <arrow/array/builder_nested.h>
+#include <arrow/chunked_array.h>
+#include <arrow/io/api.h>
+#include <arrow/result.h>
+#include <arrow/status.h>
+#include <arrow/type.h>
+#include <arrow/type_fwd.h>
+#include <arrow/type_traits.h>
```

Comment:
nit: seems like we could forward declare most things, except a couple PB classes

Reply:
100%